pkg/billing/azureblobstoragebilling.go

```diff
@@ -0,0 +1,153 @@
+package billing
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/lithammer/shortuuid"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type AzureAuthType string
+
+const (
+	// AzureAuthTypeTests is used in tests
+	// It uses well-known storage account and key.
+	// See https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite
+	AzureAuthTypeTests AzureAuthType = "tests"
+	// AzureAuthTypeDefault is for pods running in Azure Kubernetes.
+	// Make sure you have provisioned Role and you have Managed Identity.
+	AzureAuthTypeDefault AzureAuthType = "default"
```

Comment:
this auth type has not been tested yet in Azure cloud, going to do that this week before merging.

Reply:
https://github.com/neondatabase/cloud/issues/14786#issuecomment-2191013925 it worked